pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,426 @@
+        """
+        Build a string that includes the ``#include pragmas`` for the DMTF
+        schema CIM classes defined in `schema_classes` using the DMTF CIM
+        schema defined by this object.
+
+        The class names in this list can be just leaf classes. The pywbem
+        MOF compiler will search for dependent classes.
+
+        It builds a compilable MOF string in the form::
+
+            #pragma locale ("en_US")
+            #pragma include ("System/CIM_ComputerSystem.mof")
+
+        with a ``#pragma include`` for each classname in the `schema_classes`
+        list
+
+        Parameters:
+
+          schema_classes (:term:`py:list` of :term:`string` or :term:`string`):
+            These must be class names of classes in the DMTF CIM schema defined
+            for this :class:`~pywbem_mock.DMTFCIMSchema` object. This parameter
+            can be a string if there is only a single class name to be
+            included.
+
+            If the returned string is compiled, the MOF compiler will search
+            the directory defined by `schema_mof_dir` for dependent classes.
+
+        Returns:
+            :term:`string`: Valid MOF containing pragma statements for
+            all of the classes in`schema_classes`.
+
+        Raises:
+            ValueError: If any of the classnames in `schema_classes` are not in
+            the DMTF CIM schema installed
+        """
+        if isinstance(schema_classes, six.string_types):
+            schema_classes = [schema_classes]
+
+        schema_lines = []
+        with open(self.schema_mof_filename, 'r') as f:
+            schema_lines = f.readlines()
+
+        output_lines = ['#pragma locale ("en_US")\n']
+        for cln in schema_classes:
+            test_cln = '/%s.mof' % cln
+            found = False
+            for line in schema_lines:
+                if line.find(test_cln) != -1:
+                    output_lines.append(line)
+                    found = True
+                    break
+            if not found:
+                raise ValueError('Class %s not in DMTF CIM schema %s' %
+                                 (cln, self.schema_mof_filename))
+
+        return ''.join(output_lines)
+
+    def clean(self):
+        """
+        Clean the individual MOF files from the `schema_root_dir`. This removes
+        all of the individual MOF files and the mof... subdirectory for
+        the defined schema.  This is useful because while the downloaded
+        schema is a single compressed zip file, it creates several
+        thousand MOF files that take up considerable space.
+
+        The next time the :class:`~pywbem_mock.DMTFCIMSchema` object for this
+        schema_version and `schema_root_dir` is created, the zip file is
+        expanded again.
+        """
+        if os.path.isdir(self.schema_mof_dir):
+            shutil.rmtree(self.schema_mof_dir)
+
+    def remove(self):
+        """
+        If the schema has been installed it is completely removed. If
```

Comment:
Also if it was not installed but was already there.

Reply:
Rewrote to just state that if they exist.
DONE